Adapt rolling pose2d deprecated

fmrico opened the pull request:
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #5338 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | gazebo simulation of Tally |
| Does this PR contain AI generated software? | No |

---

## Description of contribution in a few bullet points

* Migrated all code to use `geometry_msgs::msg::Pose` instead of `geometry_msgs::msg::Pose2D`, which is deprecated in https://github.com/ros2/common_interfaces/pull/283
* Also fixed many CMakeLists.txt that needed `nav2_ros_common` and didn't use `target_link_libraries` for this.

## Description of documentation updates required from your changes

* I have updated doxygen/docs in the code, but not in any other place.

## Description of how this change was tested

* Pass tests with `colcon test`
* Checked functionality in `tb3_simulation_launch.py`


## Future work that may be required in bullet points

* I think there are still some problems in tests, not related to my changes, which come from newer versions of `rolling` and the compiler.

#### For Maintainers: <!-- DO NOT EDIT OR REMOVE -->
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists
- [ ] Should this be backported to current distributions? If so, tag with `backport-*`.

SteveMacenski reviewed (commented):
I got through 102/136 files, so this isn't a complete accounting but what I found so far (not much, really)

fmrico commented on nav2_controller/plugins/test/goal_checker.cpp:
```diff
@@ -53,27 +53,27 @@ void checkMacro(
+  pose0.position.x = x0;
+  pose0.position.y = y0;
+  pose0.position.z = 0.0;
+  pose0.orientation = tf2::toMsg(tf2::Quaternion({0.0, 0.0, 1.0}, theta0));
```
Ok. Replaced by `orientationAroundZAxis`

fmrico commented on nav2_dwb_controller/nav_2d_utils/CMakeLists.txt:
```diff
@@ -10,6 +10,7 @@ find_package(nav2_msgs REQUIRED)
 find_package(nav2_util REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(tf2 REQUIRED)
+# find_package(rcl_interfaces REQUIRED)
```
Ooooops, my fault. Removed..

fmrico commented:
Hi Steve,

Following your review, I have:
1. Used `orientationAroundZAxis` for creating quaternion messages.
2. Removed `nav_2d_msgs/msg/Path2D`, using `nav_msgs/msg/Path` instead.
3. Removed `nav_2d_msgs/msg/Pose2DStamped`, using `geometry_msgs/msg/PaseStamped` instead.
4. Removed conversion functions that are no longer needed because of the previous points.

SteveMacenski commented on nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp:
```diff
@@ -221,7 +224,6 @@ void BinaryFilter::process(
     }
   }
 }
-
```
Please readd this line between functions

SteveMacenski commented on nav2_route/graphs/aws_graph.geojson:
```diff
@@ -1,132 +1,1664 @@
 {
-"type": "FeatureCollection",
```
Just checking: did you update anything here or is it just reformatted from opening it in your editor?

fmrico replied:
I am not sure what happened here, because I am not aware of changing this file. I have restored it from `main` branch.

SteveMacenski commented on nav2_route/graphs/sample_graph.geojson:
```diff
@@ -1,121 +1,504 @@
 {
-    "type": "FeatureCollection",
```
Just checking: did you update anything here or is it just reformatted from opening it in your editor?

SteveMacenski commented on nav2_regulated_pure_pursuit_controller/src/collision_checker.cpp:
```diff
@@ -86,10 +86,12 @@ bool CollisionChecker::isCollisionImminent(
+  geometry_msgs::msg::Pose curr_pose;
+  curr_pose.position.x = robot_pose.pose.position.x;
+  curr_pose.position.y = robot_pose.pose.position.y;
+  tf2::Quaternion q;
+  q.setRPY(0, 0, tf2::getYaw(robot_pose.pose.orientation));
+  curr_pose.orientation = tf2::toMsg(q);
```
Questions:

1. Are there other places we should also strip out the Roll and Pitch values from and/or Poses obtained from TF that we may want to make sure we bring into 2D?
2. If so, should we make a nav2_util for this to remove boilerplate?

SteveMacenski commented on nav2_dwb_controller/dwb_critics/src/obstacle_footprint.cpp:
```diff
@@ -45,24 +45,24 @@ PLUGINLIB_EXPORT_CLASS(dwb_critics::ObstacleFootprintCritic, dwb_core::Trajector
   oriented_footprint.resize(footprint_spec.size());
-  double cos_th = cos(pose.theta);
-  double sin_th = sin(pose.theta);
+  double cos_th = cos(tf2::getYaw(pose.orientation));
```
Can we `getYaw` once and then use the theta value for the sin/cos?

SteveMacenski commented on nav2_dwb_controller/dwb_critics/src/oscillation.cpp:
```diff
@@ -172,19 +172,19 @@ void OscillationCritic::debrief(const nav_2d_msgs::msg::Twist2D & cmd_vel)
   }
   if (oscillation_reset_angle_ >= 0.0) {
-    double th_diff = pose_.theta - prev_stationary_pose_.theta;
+    double th_diff = tf2::getYaw(pose_.orientation) -
```
We should be using the angles min distance - I know this isn't a bug you introduced but this is a bug we should fix

fmrico replied:
I have added this solution:

```
  if (oscillation_reset_angle_ >= 0.0) {
    tf2::Quaternion tf2_q1, tf2_q2;
    tf2::fromMsg(pose_.orientation, tf2_q1);
    tf2::fromMsg(prev_stationary_pose_.orientation, tf2_q2);

    tf2::Quaternion delta_q = tf2_q2 * tf2_q1.inverse();
    double th_diff = tf2::getYaw(delta_q);

    if (fabs(th_diff) > oscillation_reset_angle_) {
      return true;
    }
  }
```

Let me know if you like other solution.

SteveMacenski commented on nav2_dwb_controller/dwb_plugins/src/standard_traj_generator.cpp:
```diff
@@ -208,19 +210,27 @@ nav_2d_msgs::msg::Twist2D StandardTrajectoryGenerator::computeNewVelocity(
   return new_pose;
 }
 
+
```
Remove the extra line

SteveMacenski reviewed (changes requested):
LGTM - a few small changes but I got through all the files and I should be able to just merge it in after these 

I appreciate your diligence on this. I know this isn't the most fun thing in the world

MarcoMatteoBassa commented on nav2_dwb_controller/dwb_critics/test/base_obstacle_test.cpp:
```diff
@@ -81,41 +83,45 @@ TEST(BaseObstacle, ScorePose)
+  geometry_msgs::msg::Pose pose;
+  pose.position.x = 0;
+  pose.position.y = 0;
+  pose.orientation = tf2::toMsg(tf2::Quaternion(0, 0, 0, 1));
```
Just a suggestion to save some lines: the quaternion msgs are already initialized like that, so there's no need to assign it again :)

fmrico commented on nav2_route/graphs/sample_graph.geojson:
```diff
@@ -1,121 +1,504 @@
 {
-    "type": "FeatureCollection",
```
Done.

fmrico commented on nav2_regulated_pure_pursuit_controller/src/collision_checker.cpp:
```diff
@@ -104,24 +102,27 @@ bool CollisionChecker::isCollisionImminent(
+    curr_pose.position.x += projection_time * (linear_vel * cos(theta));
+    curr_pose.position.y += projection_time * (linear_vel * sin(theta));
+    theta += projection_time * angular_vel;
+    curr_pose.orientation = curr_pose.orientation;
```
I have just checked that the correct line is:

```
    theta += projection_time * angular_vel;
    curr_pose.orientation = nav2_util::geometry_utils::orientationAroundZAxis(theta);
```

fmrico commented:
> @fmrico the `aws_graph.geojson` file is changed again, I'm going to go ahead and merge this and manually change that back for the sake of time and your sanity :-)

Thanks @SteveMacenski !!
